resources/splunk_service.rb

```diff
@@ -62,14 +65,30 @@ def install_state
   action_class do
     include CernerSplunk::ProviderHelpers
 
-    def service_action(desired_action)
+    # Provide file and service resources from methods to eliminate accidental resource cloning
+    def marker_resource
+      resources(file: marker_path.to_s)
+    rescue Chef::Exceptions::ResourceNotFound
+      file(marker_path.to_s) do
+        action :nothing
+      end
+    end
+
+    def service_resource
```

Comment:
In my testing, it was causing resource cloning due to multiple actions being run on the same service resource. I will investigate naturally created resources again in order to utilize notifications